### PR TITLE
add numeric comparison request matcher

### PR DESCRIPTION
**Pull request by armanb1989**
This PR adds numeric comparisons functionality to the state-matcher, allowing matching requests based on numeric conditions for state properties. The implementation supports comparison operators (gt, gte, lt, lte, eq, ne) with both constant values and state property values via templating.

## References

- Implements feature requested in issue #184 
- Builds upon existing state-matcher infrastructure
- Follows same patterns as other matcher types in the extension

## Submitter checklist

- [x] Recommended: Join [WireMock Slack](https://slack.wiremock.org/) to get any help in `#help-contributing` or a project-specific channel like `#wiremock-java`
- [x] The PR request is well described and justified, including the body and the references
- [x] The PR title represents the desired changelog entry
- [x] The repository's code style is followed (see the contributing guide)
- [x] Test coverage that demonstrates that the change works as expected
- [x] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

## Changes

- Added `ComparisonOperator` enum to handle numeric comparison operations
- Extended `StateRequestMatcher` to support `numericComparisons` parameter
- Added comprehensive unit tests covering all comparison operators and edge cases
- Updated documentation with usage examples and configuration details

## Testing

- Unit tests verify all comparison operators work correctly
- Tests cover both constant values and templated state property values
- Edge cases like non-numeric values and missing properties are tested
- Integration tests demonstrate real-world usage scenarios

## Documentation

Added new "Numeric comparisons match" section to README.md with:
- Supported operators reference
- Basic usage example with constant values
- Advanced example with state property templating
- Multiple property comparisons example
- Important usage notes and limitations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1052,6 +1052,113 @@ Example for nesting:
...
+    "customMatcher": {
+      "name": "state-matcher",
+      "parameters": {
+        "hasContext": "{{request.pathSegments.[1]}}",
+        "numericComparisons": {
+          "amount": {
+            "gt": "0",
+            "lte": "100"
+          }
+        }
```
Comment (dirkbolte) [inline README.md:1086]: for all other matchers, we have the following structure:

```json
"amount": {
  "contains": "100"
} 
```

applying this to your matchers, it should look like:

```json
"amount": {
  "gt": "100"
} 
```

If you can make it fit into this structure, you get `and`, `or`, `not` for free - and it would also work on list properties:

```json
"amount": {
  "and": [
    { "gt": "100" },
    { "lt": "200" }
  ]
} 
```

**Review by dirkbolte — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dirkbolte) [reply README.md:1086]: just seeing that this would require a bit more work, but it looks like a cleaner approach. I will give it a try.

Comment (armanb1989) [reply README.md:1086]: ~~Good idea! I agree, implementing the and, or, and not operators will make the system much more coherent. I will add them.~~

Comment (armanb1989) [reply README.md:1086]: I just saw your comment on the issue about implementing this logic in Wiremock core.

**Comment by dirkbolte** [comment]
The matchers were added to wiremock directly and will be available in this extension automatically when running with the right wiremock version. PR for future reference: https://github.com/wiremock/wiremock/pull/3179
